craft_parts/plugins/maven_plugin.py

```diff
@@ -106,24 +110,46 @@ def get_build_packages(self) -> set[str]:
         """Return a set of required packages to install in the build environment."""
         return set()
 
+    @property
+    def _maven_executable(self) -> str:
+        """Return the maven executable to be used for build."""
+        options = cast(MavenPluginProperties, self._options)
+        return "${CRAFT_PART_BUILD}/mvnw" if options.maven_use_mvnw else "mvn"
```

Comment:
```suggestion
        return "${CRAFT_PART_BUILD_WORK}/mvnw" if options.maven_use_mvnw else "mvn"
```

This is to handle cases with `source-subdir`, where the actual sources are in a subdirectory of a repo